server/ai_live_video.go

```diff
@@ -199,30 +198,47 @@ func startEventsSubscribe(ctx context.Context, url *url.URL, params aiRequestPar
 				continue
 			}
 
-			stream := params.liveParams.stream
-
-			if stream == "" {
-				clog.Infof(ctx, "Stream ID is missing")
-				continue
-			}
-
-			var status monitor.PipelineStatus
+			var status map[string]interface{}
 			if err := json.Unmarshal(body, &status); err != nil {
 				clog.Infof(ctx, "Failed to parse JSON from events subscription: %s", err)
 				continue
 			}
 
-			status.StreamID = &stream
+			if _, ok := status["stream_id"]; !ok {
+				clog.Infof(ctx, "Received event for stream=%s status=%+v", stream, status)
+				status["stream_id"] = stream
+				status["request_id"] = params.liveParams.requestID
+			}
 
-			// TODO: update the in-memory pipeline status
-			// monitor.UpdatePipelineStatus(stream, status)
+			monitor.SendQueueEventAsync("stream_status", status)
 
-			clog.Infof(ctx, "Received event for stream=%s status=%+v", stream, status)
+			streamStatus := map[string]interface{}{
+				"updatedAt": time.Now(),
+			}
 
-			monitor.SendQueueEventAsync(
-				"stream_status",
-				status,
-			)
+			lastStreamStatus, ok := monitor.GetStreamStatus(stream)
+			if !ok {
+				lastStreamStatus = make(map[string]interface{})
+			}
```

Comment:
**nit:** A `nil` map behaves **exactly the same** as an empty map when you are reading from it. It only explodes when writing. Given this `lastStreamStatus` is read-only (or it should be), this is redundant and can be removed.